src/modules/adaptors/Morpho/MorphoAaveV2ATokenAdaptor.sol

```diff
@@ -36,36 +36,45 @@ contract MorphoAaveV2ATokenAdaptor is BaseAdaptor, MorphoRewardHandler {
      */
     error MorphoAaveV2ATokenAdaptor__HealthFactorTooLow();
 
-    //============================================ Global Functions ===========================================
     /**
-     * @dev Identifier unique to this adaptor for a shared registry.
-     * Normally the identifier would just be the address of this contract, but this
-     * Identifier is needed during Cellar Delegate Call Operations, so getting the address
-     * of the adaptor is more difficult.
+     * @notice The Morpho Aave V2 contract on current network.
+     * @notice For mainnet use 0x777777c9898D384F785Ee44Acfe945efDFf5f3E0.
      */
-    function identifier() public pure override returns (bytes32) {
-        return keccak256(abi.encode("Morpho Aave V2 aToken Adaptor V 1.1"));
-    }
+    IMorphoV2 public immutable morpho;
 
     /**
-     * @notice The Morpho Aave V2 contract on Ethereum Mainnet.
+     * @notice The Morpho Aave V2 Lens contract on current network.
+     * @notice For mainnet use 0x507fA343d0A90786d86C7cd885f5C49263A91FF4.
      */
-    function morpho() internal pure returns (IMorphoV2) {
-        return IMorphoV2(0x777777c9898D384F785Ee44Acfe945efDFf5f3E0);
-    }
+    IMorphoLensV2 public immutable morphoLens;
 
     /**
-     * @notice The Morpho Aave V2 Lens contract on Ethereum Mainnet.
+     * @notice Minimum Health Factor enforced after every aToken withdraw.
+     * @notice Overwrites strategist set minimums if they are lower.
      */
-    function morphoLens() internal pure returns (IMorphoLensV2) {
-        return IMorphoLensV2(0x507fA343d0A90786d86C7cd885f5C49263A91FF4);
+    uint256 public immutable minimumHealthFactor;
+
+    constructor(
+        address _morpho,
+        address _morphoLens,
+        uint256 minHealthFactor,
+        address rewardDistributor
+    ) MorphoRewardHandler(rewardDistributor) {
+        _verifyConstructorMinimumHealthFactor(minHealthFactor);
+        morpho = IMorphoV2(_morpho);
+        morphoLens = IMorphoLensV2(_morphoLens);
```

Comment:
I guess for protocols that do more chain-specific customization, interfaces may not be chain-agnostic. Just something to note.

Reply:
Correct if a protocol has changed their interfaces, then we will have to make contract changes to support it no matter what we do